src/persist-client/src/operators/shard_source.rs

```diff
@@ -303,6 +306,10 @@ where
         // will block when there is no data yet available in the shard.
         cap_set.downgrade(as_of.clone());
 
+        // Wait for the start signal before ingesting any data. Do this after downgrading the
+        // capability to not unnecessarily block downstream operators.
+        start_signal.await;
```

Comment:
nit: In cases like this, I like to future proof with something like `let () = start_signal.await;`, so if the type of start_signal changes, we'll notice here with a compiler error